// @vue-macros/common@0.10.0/dist/index.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});






var _chunkONEOVRGNjs = require('./chunk-ONEOVRGN.js');








var _chunkARAOJUFSjs = require('./chunk-ARAOJUFS.js');


var _chunkY4F6JHTRjs = require('./chunk-Y4F6JHTR.js');
require('./chunk-TUFWXZYT.js');



var _chunkN5DOD465js = require('./chunk-N5DOD465.js');


var _chunk4LP5SQ45js = require('./chunk-4LP5SQ45.js');



var _chunkEYOCX5COjs = require('./chunk-EYOCX5CO.js');




















exports.DEFINE_EMITS = _chunkARAOJUFSjs.DEFINE_EMITS; exports.DEFINE_MODEL = _chunkARAOJUFSjs.DEFINE_MODEL; exports.DEFINE_OPTIONS = _chunkARAOJUFSjs.DEFINE_OPTIONS; exports.DEFINE_PROPS = _chunkARAOJUFSjs.DEFINE_PROPS; exports.DEFINE_SETUP_COMPONENT = _chunkARAOJUFSjs.DEFINE_SETUP_COMPONENT; exports.MAGIC_COMMENT_STATIC = _chunkARAOJUFSjs.MAGIC_COMMENT_STATIC; exports.MagicString = _chunk4LP5SQ45js.MagicString; exports.REPO_ISSUE_URL = _chunkARAOJUFSjs.REPO_ISSUE_URL; exports.addToScript = _chunkEYOCX5COjs.addToScript; exports.babelParse = _chunkONEOVRGNjs.babelParse; exports.checkInvalidScopeReference = _chunkONEOVRGNjs.checkInvalidScopeReference; exports.finalizeContext = _chunkN5DOD465js.finalizeContext; exports.getLang = _chunkONEOVRGNjs.getLang; exports.getTransformResult = _chunkY4F6JHTRjs.getTransformResult; exports.initContext = _chunkN5DOD465js.initContext; exports.isCallOf = _chunkONEOVRGNjs.isCallOf; exports.isLiteralType = _chunkONEOVRGNjs.isLiteralType; exports.isStaticExpression = _chunkONEOVRGNjs.isStaticExpression; exports.parseSFC = _chunkEYOCX5COjs.parseSFC;
